src/Button/Button.js

```diff
@@ -10,13 +10,13 @@ const Button = React.forwardRef(
   ({ children, iconStart, iconEnd, spin, variant, type, ...props }, ref) => (
     <StyledButton
       {...{
-        ...props,
-        iconStart,
+        className: `button--${variant}`,
         iconEnd,
+        iconStart,
         ref,
         type,
         variant,
-        className: `button--${variant}`
+        ...props
```

Comment:
This was the main issue. I imagine it stems from something that happens with the ref, but the CSS was never being overwritten.